Lib/asyncio/unix_events.py

```diff
@@ -394,6 +394,11 @@ def _sock_sendfile_native_impl(self, fut, registered_fd, sock, fileno,
                 fut.set_result(total_sent)
                 return
 
+        # On 32-bit architectures truncate to 1GiB to avoid OverflowError,
+        # see bpo-38319/gh-82500.
+        if sys.maxsize < 2 ** 32:
+            blocksize = min(blocksize, 2 ** 30)
```

Comment:
I don't feel the comment needs to reference bpo/gh. It should just explain what we are doing and why. The first line of the comment says it all. Historians can look through the git commit history.

Separately, the check for sys.maxsize feels odd -- especially since it's not obvious that maxsize couldn't be much smaller still, in which case 2**30 might still be too large. :-) Maybe `sys.maxsize//2 + 1` would work instead of hardcoding 2**30? You could just write, unconditionally,
```py
blocksize = min(blocksize, sys.maxsize//2 + 1)
```

Reply:
It seems like that value `2 ** 30` is safe based on 938c00ca9e4207a2531041edff2e82490b02047f and the history in https://bugs.python.org/issue38319 and Home Assistant has been running this patch for a while now

Reply:
```
#ifdef HAVE_LARGEFILE_SUPPORT
    *((Py_off_t *)addr) = PyLong_AsLongLong(arg);
#else
    *((Py_off_t *)addr) = PyLong_AsLong(arg);
#endif
```

Reply:
https://github.com/python/cpython/blob/7fc9be350af055538e70ece8d7de78414bad431e/Modules/posixmodule.c#L10953

Reply:
I don't see a way to check `HAVE_LARGEFILE_SUPPORT` in python code

Reply:
`blocksize = min(blocksize, sys.maxsize//2 + 1)` seems better to me after working through it.

Reply:
bde4502f12fe3dfff4c230da73cd67bfe7b9e981

Reply:
Great!